overlord/snapstate/snapmgr.go

```diff
@@ -660,18 +657,19 @@ func snapSetupAndState(t *state.Task) (*SnapSetup, *SnapState, error) {
 
 func (m *SnapManager) undoMountSnap(t *state.Task, _ *tomb.Tomb) error {
 	t.State().Lock()
-	ss, snapst, err := snapSetupAndState(t)
+	ss, _, err := snapSetupAndState(t)
 	t.State().Unlock()
 	if err != nil {
 		return err
 	}
 
-	pb := &TaskProgressAdapter{task: t}
-	typ, err := snapst.Type()
```

Comment:
ah, I see the problem (snapst doesn't store itself anymore), we should stick type on the task